src/main/java/frc/trigon/robot/subsystems/swerve/Swerve.java

```diff
@@ -0,0 +1,78 @@
+package frc.trigon.robot.subsystems.swerve;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Swerve extends SubsystemBase {
+    private final static Swerve INSTANCE = new Swerve();
+
+    public Swerve() {
+        zeroHeading();
+    }
+
+    public static Swerve getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Drives swerve relitive to itself.
+     *
+     *  @param translation disired speed in meters in the two axes.
+     *   @param rotation sets the angle of the robot in radioants.
+     **/
+    void selfRelativeDrive(Translation2d translation, Rotation2d rotation) {
+        ChassisSpeeds chassisSpeeds = new ChassisSpeeds(
+                translation.getX(),
+                translation.getY(),
+                rotation.getRadians()
+        );
+        selfRelative(chassisSpeeds);
+    }
+
+    /**
+     * Drives swerve reltive to the field.
+     *
+     * @param translation disired speed in meters in the two axes.
+     * @param rotation sets the angle of the robot in radioants.
+     **/
```

Comment:
same comments from the self relative documentation

Reply:
wdum

Reply:
the same comments that I give for the documentation of the function above probably also apply here